### ISSUE TITLE
Slow game downloads

### ISSUE THREAD

**Issue by AnKosteck** [flathub/com.valvesoftware.Steam#336]
My problem is a strange one: I'm running [KaOS](https://kaosx.us/) on a Lenovo Thinkpad E485 and installed the Flatpak-Steam. Where I live however all game downloads happen at a few 100KB/s. The Steam client and the Flatpak dependencies update at full speed though.

The type of internet connection, wireless or hardwired or both, does not matter since the problem persists with any. I testet the download speeds at 2 additional and different locations and there I could get the games at full speed each. This problem only occurs at my home via that net provider (A1 in Austria).

I testet the game downloads with a separate machine running Windows just now and the problem does not appear there either. Thus this problem is limited to my laptop running KaOS and the Flatpak-Steam client via the modem of my internet provider.

So I am guessing that the Flatpak-Steam does something differently than the non-Flatpak Steam client and this is capped or limitted via the provider/modem. Or maybe a bad [configure](https://github.com/KaOSx/apps/blob/master/flatpak/PKGBUILD) flag was set ? Does anyone have a hint on this?

**Comment by nanonyme** [comment]
I have no idea what could be happening. Maybe Steam is somehow confused by your ISP's location and is picking unoptimal mirrors when running inside Flatpak context?

**Comment by AnKosteck** [comment]
Thanks for your answer.
I tried a few things the last days starting with observing the traffic on the ISP's supplied modem. What I noticed was that the Flatpakt-Steam client seemed to stay on 1-4 connections to Valve servers whereas another machine running Windows seems to open and close many connections throughout the download (if this makes sense). The chosen servers had a suitable location though.

Then I tried Fedora 29 on my Thinkpad and expected the Steam client from rpmfusion to work without hitches which was not the case. Both clients, rpmfusion and Flatpak-Steam had slow download speeds. Which was strange since my previous machine did not have this problem under Fedora. So I installed Windows and there Steam worked flawlessly. So I suspected some strange IOMMU/kernel issue since the Thinkpad has a broken ivrs table.
I then proceeded to test various IOMMU settings and other kernel arguments but to no prevail.
I then searched for similar Steam problems on Linux and may have found the solution. 


Long story short, this issue seems to be DNS related. The fix for me was to install bind locally and set named to act as a caching and forwading server. With that I have high download rates with both the Steam clients installed from rpmfusion and Flathub.

So case closed I guess?

**Comment by nanonyme** [comment]
I dunno. I'm a bit disappointed if it's DNS. Let's keep this open for now. Do you happen to know which hostnames this was about?

**Comment by AnKosteck** [comment]
Sorry for the delay, I had no time for gaming and lost track of this. The issue was most likely the modem our ISP provided.
After switching to a different provider and with a newer modem all works as intended now.
